documentation/contributing/pullrequests.markdown

```diff
@@ -0,0 +1,85 @@
+---
+author: v4hn
+comments: true
+date: 2016-8-3 13:24:00+00:00
+layout: page
+slug: pullrequests
+title: Pullrequests
+---
+
+# Handling Contributions
+
+MoveIt is a huge project with multiple maintainers and many contributing users.
+Thus, sometimes it becomes quite hard to keep track of all open requests and their current state.
+To ease the process of getting your contribution merged into the project, the following provides a guideline for contributors as well as maintainers.
+
+## Contributor/User Side
+
+Your pull-requests might target any branch of a MoveIt repository that supports an [OSRF-supported ROS distribution](http://wiki.ros.org/Distributions).
+The most important point about your request is that you verified it is working. Obviously, the best way to do that is with the setup/distribution you use.
+
+### keep requests clean and independent
+
+Different commits in a request should correspond to logical units that build up on each other. Feel free to rebase and force-push your feature-branch to address feedback.
+If your request consists of multiple commits with titles like "cleanup braces" or "address xyz's feedback", the maintainer who merges the request will squash-merge and your changes will end up as a single commit in the project.
```

Comment:
Should we also recommend they not do that, instead just always amending? Here would be a good place to add a tip for Git newbies about how to do this:

> git rebase -i HEAD~1 

or 

> git commit -a --amend